### ISSUE TITLE
Feather: Scopes are mangled in the implicit global scope, resulting in incorrect Code Completion suggestions

### ISSUE THREAD

**Issue by jack27121** [YoYoGames/GameMaker-Bugs#2289]
### Description
Say you make a constructor for vector2 

```
function Vector2(_x, _y) constructor
{
    x = _x;
    y = _y;

    //@function Add()
    static Add = function(_vec2)
    {
        x += _vec2.x;
        y += _vec2.y;
    }
} 

myvec = new Vector2(1,4);

myvec.Add() //this should have jsdoc help text, and autocomplete
Add() //this definetely should not, but it appears.
```

If you have a function called Add, and you also have a constructor including a function with the same name, the jsdoc text conflict even though it shouldn't

### Steps To Reproduce
within the included project, go to obj_collision 's draw event,
and check the last 2 lines. Autocomplete happens for get_point() even tho it only should appear when trailing a variable made from the constructor
and the helper text shouldn't appear either.

[61ee7797-4315-4ce3-80c2-d1c75c5187eb](https://api.gamemaker.io/api/github/downloads/61ee7797-4315-4ce3-80c2-d1c75c5187eb)

**Comment by jack27121** [comment]
maybe this is more of a feature request, i don't know. But regardless many people would like this to work easier, especially for extention creators that are unable to make descriptive jsdoc for their functions since it will conflict with whatever project it's included in

**Comment by Totobal5** [comment]
Oh please fix this is so bad when importing libraries

**Comment by zreedy** [comment]
Partially fixed.
There are 3 bugs reported in this ticket:
 - FIXED: Variables, and Static Functions declared inside of a Constructor Function visible in the implicit global scope.
 - JSDoc does not parse on `static ident = function()` statements
 - Implicit global instance variables mangle their internal type reference, resulting in incorrect auto-complete on `.` dereference

**Comment by zreedy** [comment]
I have updated the title of this ticket to more accurately reflect the issues described.
Additionally, the provided code snippet incorrectly specified `add` and `vector2` which Feather _correctly_ treats as `Any` types as the symbol name was typo'd. The issue described can be observed against the updated code snippet.

**Comment by zreedy** [comment]
All issues now fixed, however the issues of the implicit global instance variables mangling their types is the result of a Type System limitation. There's a workaround the address the code completion issue described here, but the color coding issues are not addressable at this time.

**Comment by scott-dunbar** [comment]
Verified fixed in Beta IDE v2024.1100.0.626 Beta Runtime v2024.1100.0.652
